test/v3/controllers/requestParsers/validators/SubmitSelfEmploymentBsasValidatorSpec.scala

```diff
@@ -57,117 +61,127 @@ class SubmitSelfEmploymentBsasValidatorSpec extends UnitSpec {
+          List(CalculationIdFormatError)
       }
-      "passed an empty body" in {
-        validator.validate(SubmitSelfEmploymentBsasRawData(nino, bsasId, AnyContentAsJson(Json.obj()))) shouldBe List(RuleIncorrectOrEmptyBodyError)
+    }
+
+    "return RuleIncorrectOrEmptyBodyError" when {
+      "an empty body is submitted" in {
+        validator.validate(SubmitSelfEmploymentBsasRawData(nino = nino, calculationId = calculationId, AnyContentAsJson(Json.obj()))) shouldBe List(
+          RuleIncorrectOrEmptyBodyError)
       }
-      "passed an invalid body" in {
-        validator.validate(SubmitSelfEmploymentBsasRawData(nino, bsasId,
-          AnyContentAsJson(Json.obj("income" -> "beans")))) shouldBe List(RuleIncorrectOrEmptyBodyError.copy(paths = Some(Seq("/income"))))
-        validator.validate(SubmitSelfEmploymentBsasRawData(nino, bsasId,
-          AnyContentAsJson(Json.obj("income" -> "{}")))) shouldBe List(RuleIncorrectOrEmptyBodyError.copy(paths = Some(Seq("/income"))))
-        validator.validate(SubmitSelfEmploymentBsasRawData(nino, bsasId,
-          AnyContentAsJson(Json.toJson(defaultBody(income = Some(Income(None, None))))))) shouldBe List(RuleIncorrectOrEmptyBodyError)
-        validator.validate(SubmitSelfEmploymentBsasRawData(nino, bsasId,
-          AnyContentAsJson(Json.obj()))) shouldBe List(RuleIncorrectOrEmptyBodyError)
-        validator.validate(SubmitSelfEmploymentBsasRawData(nino, bsasId,
-          AnyContentAsJson(mtdRequestWithBothExpenses))) shouldBe List(RuleBothExpensesError)
-        validator.validate(SubmitSelfEmploymentBsasRawData(nino, bsasId,
-          AnyContentAsJson(mtdRequestWithAdditionsAndExpenses))) shouldBe List(RuleBothExpensesError)
+
+      "object is empty" in {
+        validator.validate(SubmitSelfEmploymentBsasRawData(
+          nino,
+          calculationId,
+          AnyContentAsJson(Json.obj("income" -> "{}")))
+        ) shouldBe List(RuleIncorrectOrEmptyBodyError.copy(paths = Some(Seq("/income"))))
+      }
+
+      "fields are empty" in {
+        validator.validate(SubmitSelfEmploymentBsasRawData(
+          nino,
+          calculationId,
+          AnyContentAsJson(Json.toJson(defaultBody(income = Some(Income(None, None))))))
+        ) shouldBe List(RuleIncorrectOrEmptyBodyError)
+      }
+
+      "object is invalid" in {
+        validator.validate(SubmitSelfEmploymentBsasRawData(
+          nino,
+          calculationId,
+          AnyContentAsJson(Json.obj("income" -> "beans")))
+        ) shouldBe List(RuleIncorrectOrEmptyBodyError.copy(paths = Some(Seq("/income"))))
+      }
+    }
+
+    "return RuleBothExpensesError" when {
+      "both expenses are present" in {
+        validator.validate(SubmitSelfEmploymentBsasRawData(
+          nino,
+          calculationId,
+          AnyContentAsJson(mtdRequestWithBothExpenses))
+        ) shouldBe List(RuleBothExpensesError)
       }
+    }
+
+    "return ValueFormatError" when {
+      "multiple fields are invalid" in {
```

Comment:
Need tests for single fields being invalid too. 

(See `SubmitForeignPropertyBsasValidatorSpec` for one way to do this relatively compactly. Basically have some JSON with everything in it and update each field in turn to bad values using methods in the `JsonErrorValidators` trait.)